src/core/dagservice.ts

```diff
@@ -83,14 +84,22 @@ export class DAGService {
    * @param {string} path The path that should be resolved.
    */
   async *resolve(cid: CID, path: string): AsyncIterableIterator<ResolveResult> {
-    const data = await this.blockService.get(cid)
-    const block = Block.create(data.data, data.cid)
-    const reader = block.reader()
-    const node = reader.get(path)
-    yield { value: node.value, remainderPath: node.remaining || '' }
-    if (CID.isCID(node.value)) {
-      for await (const res of this.resolve(node.value, node.remaining || '')) {
-        yield res
+    const data = await this.blockService.get(cid);
```

Comment:
Some new stuff in here to see. Mostly just dealing with the new Block API updates.